File: notification-service/Events/WorkspaceUserAddedEvent.cs
using System;
using System.Collections.Generic;
using System.Text;

namespace NotificationService.Events
{

    public class WorkspaceUserAddedEvent
    {
        public Guid EventId { get; set; }

        public long WorkspaceId { get; set; }

        public string WorkspaceName { get; set; } = string.Empty;

        public long AddedUserId { get; set; }

        public string AddedUserName { get; set; } = string.Empty;

        public string AddedUserEmail { get; set; } = string.Empty;

        public string Role { get; set; } = string.Empty;

        public string AddedBy { get; set; } = string.Empty;

        public DateTime OccurredOn { get; set; }
    }
}

File: workspace-service/src/main/java/com/teamcollab/workspaceservice/workspace/event/WorkspaceUserAddedEvent.java
package com.teamcollab.workspaceservice.workspace.event;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

import java.time.LocalDateTime;
import java.util.UUID;

@Data
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class WorkspaceUserAddedEvent {

    private UUID eventId;

    private Long workspaceId;

    private String workspaceName;

    private Long addedUserId;

    private String addedUserName;

    private String addedUserEmail;

    private String role;

    private String addedBy;

    private LocalDateTime occurredOn;

}
